src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py

```diff
@@ -168,6 +169,9 @@ def collectAgentInfo(self):
             if float(disk['percent'].strip('%')) >= diskUseThreshold and \
                             disk['mounted'] not in self.config.AnalyticsDataCollector.ignoreDisk:
                 agentInfo['disk_warning'].append(disk)
+                agentInfo['drain_mode'] = True
+                self.config.WorkQueueManager.queueParams['DrainMode'] = True
+                os.system("$manage execute-agent wmcoreD --restart --components=WorkQueueManager,AnalyticsDataCollector,AgentStatusWatcher")
```

Comment:
Sorry for the belated answer. `self.agentInfo` just adds a couple of key/value pairs which are later on uploaded to wmstats.

And looking at that code, it doesn't really update the configuration every cycle, since it uses the same config object that is passed when starting that thread :-(

We could use this function https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Configuration.py#L573 for reloading the agent config, but then how do we know the filename path? We can discuss it in the meeting.